toolkit/tools/imagegen/configuration/systemconfig.go

```diff
@@ -138,6 +140,19 @@ func (s *SystemConfig) IsValid() (err error) {
 		return fmt.Errorf("invalid [KernelCommandLine]: %w", err)
 	}
 
+	// Validate that PackageRepos do not contain duplicate package repo name
+	repoNames := make(map[string]bool)
+	for _, packageRepo := range s.PackageRepos {
+		if err = packageRepo.IsValid(); err != nil {
+			return fmt.Errorf("invalid [PackageRepo]: %w", err)
+		}
+
+		if _, ok := repoNames[packageRepo.Name]; ok {
```

Comment:
Nit: we can simplify a bit, if you're interested:
```suggestion
		if repoNames[packageRepo.Name] {
```

If the value is not present in the map, it will be created with the default value for `bool`, which is `false`. Since we update the map immediately if this check fails, we don't really create unnecessary entries.